nucypher/blockchain/eth/actors.py

```diff
@@ -381,16 +380,25 @@ def locked_tokens(self):
         return self.miner_agent.get_locked_tokens(miner_address=self.checksum_public_address)
 
     @property
-    def total_staked(self) -> int:
+    def total_staked(self) -> NU:
         if self.stakes:
-            return sum(stake[-1] for stake in self.stakes)
-        return 0
+            return NU(sum(int(stake.value) for stake in self.stakes.values()), 'NuNit')
+        return NU(0, 'NuNit')
+
+    def __read_stakes(self) -> None:
+        stakes_reader = self.miner_agent.get_all_stakes(miner_address=self.checksum_public_address)
+        stakes = dict()
+        for index, stake_info in enumerate(stakes_reader):
+            stake = Stake.from_stake_info(owner_address=self.checksum_public_address,
+                                          stake_info=stake_info,
+                                          index=index)
+            stakes[index] = stake
+        self.__stakes = stakes
 
     @property
-    def stakes(self) -> Tuple[list]:
+    def stakes(self) -> dict:
         """Read all live stake data from the blockchain and return it as a tuple"""
-        stakes_reader = self.miner_agent.get_all_stakes(miner_address=self.checksum_public_address)
-        return tuple(stakes_reader)
+        return self.__stakes
```

Comment:
I think, at this point it's not live stake but cached stake

Reply:
:heavy_check_mark: